### PR TITLE
Added filename as an anchor parameter for BitBucket line link support.

### PR DESCRIPTION
**Pull request by dharmabruce**
Sample link to a line in a file of a bitbucket repository:
https://bitbucket.org/fboender/bbcloner/src/master/src/bbcloner#bbcloner-32

### CODE REVIEW

**Comment by jklein** [comment]
I assume this is tested and works with anchor patterns that have no filename? If so, then it looks good to me. @kellegous what do you think?

**Comment by dharmabruce** [comment]
Yes, I've tested it with anchor patterns that have no filename and paths with single to several levels of directories.

File: ui/assets/js/common.js [hunk 1]
```diff
@@ -13,7 +13,8 @@ var lib = {
     UrlToRepo: function(repo, path, line, rev) {
         var url = repo.url.replace(/\.git$/, ''),
             pattern = repo['url-pattern'],
-            anchor = line ? lib.ExpandVars(pattern.anchor, { line : line }) : '';
+            filename = path.replace(/^.*\//, ''),
```
Comment (kellegous) [inline ui/assets/js/common.js]: I don't understand the intent of this line. Is this to strip the trailing "/"?

Comment (kellegous) [inline ui/assets/js/common.js]: Oh I see, this is extracting the basename. Seems like there is a more straightforward approach with `lastIndexOf`. `path.substring(path.lastIndexOf('/')+1)`?

Comment (dgryski) [inline ui/assets/js/common.js]: http://golang.org/pkg/path/#Base ?

**Comment by dharmabruce** [comment]
I agree that the suggestion clarifies the intention. I will push an update today.

File: ui/assets/js/common.js (hunk 1 above)
Comment (dharmabruce) [inline ui/assets/js/common.js]: I agree that the suggestion clarifies the intention. I will push an update today.

**Comment by davidmankin** [comment]
I've tested this and it works for me.  Would be great to see it merged.

**Comment by kellegous** [comment]
Sorry for the delay.

**Pull request merged**